src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1167,43 +1165,37 @@ private boolean confirmClose(LibraryTab libraryTab) {
     /**
      * Ask if the user really wants to remove any empty entries
      */
-    private Boolean confirmEmptyEntry(LibraryTab libraryTab, BibDatabaseContext context) {
+    private void confirmEmptyEntry(LibraryTab libraryTab, BibDatabaseContext context) {
         String filename = libraryTab.getBibDatabaseContext()
                                     .getDatabasePath()
                                     .map(Path::toAbsolutePath)
                                     .map(Path::toString)
                                     .orElse(Localization.lang("untitled"));
 
-        ButtonType deleteEmptyEntries = new ButtonType(Localization.lang("Delete empty entries"), ButtonBar.ButtonData.YES);
-        ButtonType keepEmptyEntries = new ButtonType(Localization.lang("Keep empty entries"), ButtonBar.ButtonData.NO);
-        ButtonType cancel = new ButtonType(Localization.lang("Return to JabRef"), ButtonBar.ButtonData.CANCEL_CLOSE);
+        if (prefs.getGeneralPreferences().shouldConfirmDeleteEmptyEntries()) {
+           boolean response = dialogService.showConfirmationDialogWithOptOutAndWait(
+                    Localization.lang("Empty entries"),
+                    Localization.lang("Library '%0' has empty entries. Do you want to delete them?", filename),
+                    Localization.lang("Delete"),
+                    Localization.lang("Cancel"),
+                    Localization.lang("Do not ask again"),
+                    (optOut) -> prefs.getGeneralPreferences().setConfirmDeleteEmptyEntries(!optOut));
+           prefs.getGeneralPreferences().setDeleteEmptyEntries(response);
```

Comment:
This changes the user's preferences. The PR did not show a screenshot of that setting.

If I (as user) am asked whether I want to delete empty entries, what happens the next time I press on save? Is my last setting preserved?

I this is used only for the current call of write, this should be done differently (maybe as it was before)

Reply:
Hi @koppor , I am not too sure about your question.
1. If user does not tick on "Do not ask again" checkbox, system will keep popping the confirmation dialog about empty entries deletion.
2. If user does tick on "Do not ask again" checkbox, `prefs.getGeneralPreferences().setConfirmDeleteEmptyEntries` used to store the value and no longer show the confirmation dialog in future
3. If user clicks on "Delete" button and tick on "Do not ask again" checkbox, `prefs.getGeneralPreferences().setDeleteEmptyEntries(response)` **always** delete empty entries in future.
4. If user does not click on "Delete" button but tick on "Do not ask again" checkbox, `prefs.getGeneralPreferences().setDeleteEmptyEntries(response)` **always does not** delete empty entries in future.

Reply:
Thank you for the explanation. I read that the preference is **always** saved, independent of the checkbox "Do not ask again". This is IMHO the first occasion where during a usage of JabRef the preferences are changed. (Side note: It is OK for me, because I used that concept a decade ago. However, we removed that approx. 5 years ago not to confuse users)

With "did not show a screenshot" I meant that "Always delete entries" is not showing up at https://user-images.githubusercontent.com/49628911/167340402-3bde988e-d67e-4734-bf8f-3b4992ed219d.png - or do I miss something?

Reply:
Thank you for your review @koppor.
In my humble opinion, "Always delete entries" preference relies on the empty entries deletion confirmation dialog. Therefore, I have only added "Show confirmation dialog when deleting empty entries" for user to access the preference of confirmation dialog.
Could you please advice if "Always delete entries" is needed at https://user-images.githubusercontent.com/49628911/167340402-3bde988e-d67e-4734-bf8f-3b4992ed219d.png?

Reply:
@koppor  Not sure, we have a couple of confirmation dialogs where the choice is stored in the preferences